src/server/config.rs

```diff
@@ -2,36 +2,41 @@
 
 use std::path::PathBuf;
 
-use crate::bsp_types::BuildClientCapabilities;
+use crate::bsp_types::requests::BuildClientCapabilities;
+use crate::logger::log;
 use crate::project_model::ProjectManifest;
 
 #[derive(Debug, Clone)]
 pub struct Config {
-    pub discovered_projects: Vec<ProjectManifest>,
+    pub workspace_manifest: ProjectManifest,
     pub caps: BuildClientCapabilities,
     root_path: PathBuf,
 }
 
 impl Config {
     pub fn new(root_path: PathBuf, caps: BuildClientCapabilities) -> Self {
-        Config {
-            discovered_projects: vec![],
+        let mut this = Config {
+            workspace_manifest: ProjectManifest::default(),
             caps,
             root_path,
-        }
+        };
+        this.update_project_manifest();
+        this
     }
 
     pub fn root_path(&self) -> &PathBuf {
         &self.root_path
     }
-}
 
-impl Config {
-    pub fn linked_projects(&self) -> Vec<ProjectManifest> {
-        self.discovered_projects
-            .iter()
-            .cloned()
-            .map(ProjectManifest::from)
-            .collect()
+    pub fn update_project_manifest(&mut self) {
+        match ProjectManifest::discover(&self.root_path) {
+            Ok(workspace_manifest) => {
+                self.workspace_manifest = workspace_manifest;
+            }
+            Err(e) => {
+                log(&format!("error: {}", e));
+                todo!() // Add Logging to client and change server state to waiting for reload
```

Comment:
swoją drogą, można dodawać komentarze wewnątrz todo
```suggestion
                todo!("Add Logging to client and change server state to waiting for reload")
```